addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/handler/OpenDaikinAcUnitHandler.java

```diff
@@ -0,0 +1,264 @@
+                                thing.getUID().getAsString(), channelUID.getId());
+                    }
+                    break;
+            }
+
+            poll();
+        } catch (OpenDaikinCommunicationException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OpenDaikin AC Unit");
+        OpenDaikinConfiguration config = getConfigAs(OpenDaikinConfiguration.class);
+        if (config.host == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Host address must be set");
+        } else {
+            webTargets = new OpenDaikinWebTargets(client, config.host);
+            refreshInterval = config.refresh;
+
+            schedulePoll();
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private synchronized void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to Daikin controller", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to Daikin controller", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void pollStatus() throws IOException {
+        ControlInfo controlInfo = webTargets.getControlInfo();
+        updateStatus(ThingStatus.ONLINE);
+        if (controlInfo != null) {
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+            if (controlInfo.temp.isPresent()) {
+                updateTempState(OpenDaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp.get());
+            }
+
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+            updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
```

Comment:
Hello @caffineehacker ,

I set up the development enviroenment and took a look a the code made a change and tested with a good final result for the update of the fan swing. Could you please consider in appying the change and adding the line of code to the binding?

updateState(OpenDaikinBindingConstants.CHANNEL_AC_FAN_DIR, new StringType(controlInfo.fanMovement.name()));

Thanks.

Reply:
Sorry for the very slow response. I just applied your fix and pushed to GitHub. A new build should be available from travis-ci soon (which is automatically what the Eclipse Marketplace serves up). Let me know if that solves the issue.

Reply:
Thanks for the update. I have been testing this modification since I did the change and generated the jar file. As soon as your build is available I will install from the marketplace and "take it for a spin".